### ISSUE TITLE
Spike: Training Tools within Drupal CMS 

### ISSUE THREAD

**Issue by gracekretschmer-metrostar** [department-of-veterans-affairs/va.gov-cms#19164]
## User Story or Problem Statement
As an editor of Drupal CMS, I need more support and training to better understand how to quickly understand to do what they need to do. 

As a CMS PO, I need to understand training tools that is feasible knowing the staffing and skill constraints that are on the CMS team. 

## Description or Additional Context 
Editor Survey indicated that editors want more support in training and that editors are only spending 10% of their workday in Drupal CMS. Erika has been looking to a learning management tool that would possibly integrate into TMS and identified that we do have some funding to invest in a LMS. Before we decide how to move forward on if/how we leverage any training tools, we need an UX and accessibility evaluation for the best tools to meet editors' needs, while contrasting that to the budgetary, skill, and capacity constraints on the team. 

## Supporting User Research
- https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/cms/research/knowledge-base/kb-research-findings.md

## Steps for Implementation
- Explore Drupal modules that could replicate FETU or tooltips
- Explore the user agreement that the Teams Site workflow
- Explore learning management options and opportunities to integrate into TMS. 
- Consult with Michelle Middaugh with her insight in training support for editors. 

## Acceptance Criteria
- [ ] Documentation on what options are available that includes level of effort, skills required for implementation and maintaining tool, costs, and impact to users.

**Comment by srancour** [comment]
Documented in a [Confluence document](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/3475505158/Training+tools+for+editors). For those who don't have access to Confluence here is the breakdown of this work:

## TLDR
I think multiple things here would be best to used together. I think the highest values to be used together would be [Tour](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/edit-v2/3475505158#Tour), the [Changelog](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/edit-v2/3475505158#Changelog) ideas, [Editor contributions](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/edit-v2/3475505158#Editor-contributions-in-KB), and the [Glossary of terms](https://vfs.atlassian.net/wiki/spaces/PCMS/pages/edit-v2/3475505158#Glossary-of-terms).

## First time user experience
It would be nice to have a first time user experience that can walk users through how to use Drupal and could be pulled up afterwards in case they forget.

### Tour
[Tour](https://www.drupal.org/project/tour) is a module that could add in guided tours that we could build ourselves. It could also potentially work for the tooltip idea as well. The selector can be placed on any element via class, selector, and/or id. It could be a good starting point, but it does seem to have some issues with accessibility (e.g. sets focus on the previous button instead of the heading on change, doesn’t give a screen reader user any idea on what it’s pointing to, can’t close modal by clicking outside) and some odd UX bugs like the arrow highlight not lining up properly. It also became deprecated by core and went into contrib so there has been a lot of iterations as of late in the 2.x release.

#### Pros:

- Can build entire flows that can help directly in the flow the user is doing
- Great way of helping to train new users
- Reusable, users can pull it back up if they forget
- Very little dev work once up and running
- Can be a good starting point

#### Cons:

- Becomes something that would need to be updated as changes happen
- Some dev work would be needed to address accessibility issues in the module
- No longer part of Drupal Core

## Tooltip
It would be nice to have tooltips that an editor can press to better understand what something is while in context.

### Toggletip
[Toggletip](https://www.drupal.org/project/drupal/issues/3197758) is a new feature that is being worked on, but does seem to be close to releasing inside Drupal. The feature allows for information icons to appear on elements that has a message that opens when pressed.

#### Pros:

- Add extra context directly in the flow the user is in
- Can add links to the KB articles for more information

#### Cons:
- Not yet released in Drupal
- Only adds extra context if the user knows to use it
- Could need to be updated when elements are updated

## Loop linking
It would be nice if in the feature that the editor is in we had links to the KB articles about how to use it. On the other end, it would be nice in the KB to have a link to the feature.

## LMS Style tools
### TLDR:
I would not suggest LMS style tools for editor training. The amount of overhead for our team would be immense enough that you could hire an entire other team to manage.

Issue with all LMS tools is the upkeep from our team to make sure they are consistently updated to what has changed in Drupal. With Drupal constantly updating and all CMS teams adding more functionality and working to make things better, this becomes a never ending chasing our tail to get new videos in and trainings updated. LMS tools are also now a dime a dozen so finding one that the VA doesn’t already have becomes a lengthy vetting process not only for functionality, but also for security purposes.

### Loom
[Loom](https://www.loom.com/) could be used to make video tutorials for editors. We could use these for one off quick tutorials in the KB, but maintaining with new updates to the flows becomes a time sink. Free accounts allow for 20 videos per user which can be up to 5 minutes long. Paid plans are:

- Business
   - $15/seat/month
   - Unlimited videos
   - Unlimited recording time
   - Basic editing
   - Remove loom branding
   - Upload and download
   - Business + AI
   - $20/seat/month
   - Filler word removal
   - Auto titles and summaries
   - AI workflows
   - Edit by transcript
   - Variables
   - More advanced editing
- Enterprise
   - No listed price
   - SSO and SCIM
   - Advanced privacy
   - Custom data policies
   - Zoom integration
   - Salesforce integration
   - Request email to view

#### Pros:
- Useful for creating quick videos for walkthroughs of how to do common tasks

#### Cons:
- Cost to be able to do more than 20 videos
- Cost to be able to download videos
- Lots of overhead for our team to create and maintain
- No certifications or testing

### TMS
Internal tool that VA already has and employees have to use for annual training and can use for continuing ed.

#### Pros:
- License already owned
- Editors already use it for other training
- Flexible in style of learning
- Video and text options
- Can test user’s knowledge
- Can set requirements for re-learning
- Can give out certificates

#### Cons:
- Lots of overhead for our team to create and maintain the course(s)
- Can only access on internal VA network

## Changelog
It would be nice to have a changelog that, when they log in, shows the editor everything that has changed within Drupal since the last time they logged in so they can better understand if the flow they used has been altered in any way.

### Changelog preview
There is a module called [Changelog preview](https://www.drupal.org/project/changelog_preview) that would make it so we have to maintain the changelog via markdown. It can’t pop up for the user to see, but it does create a changelog link that goes to a page that can be configured and can have permissions set on it. Adds lots of overhead for our team to maintain. Could be added into collab cycle to update?

#### Pros:
- Can be added to Drupal and integrated
- Would give editors something they have been asking for
- Can pull from multiple md files so we could have each team maintain their own changelog or set it up to be per user type (e.g. all, VAMC, VBA, etc.)

#### Cons:
- Users need to go elsewhere instead of directly inside the flow
- Teams would need to add a task to update their changelog when they make changes

### CMS Announcements
We do have a [CMS Announcements](https://prod.cms.va.gov/help/cms-announcements) page that does exist, however it has been abandoned.

#### Pros:
- Already exists in Drupal
- No extra tools to edit and maintain
- Editing is done directly in Drupal

#### Cons:
- Users would need to know to go and look at the page
- No easy way of letting users know there has been a change
- Extra page that would need to be maintained by multiple teams
- Hard to train users on new things through a changelog except for linking to other things

## Editor contributions in KB
### Allowing comment boxes in KB
It would be nice to have a way that editors can comment on KB articles to help each other, offer suggestions for updating KB articles, giving tips for how they do something.

#### Pros:
- Editors can help each other
- Editors can give their own tips on how to do things
- Editors can get us feedback on how to improve the KB as a whole in real time as they are using it
- Already built into Drupal, just need to turn on for KB pages

#### Cons:
- Editors would only be able to use this in the KB and it would not help directly in the flow
- Lots of comments could be added as more editors comment and useful information could be lost

## User agreement for editors
It would be nice to have a user agreement that editors sign yearly saying that they know how to do the work they need to do and have done the training.

## Agreement editor knows what they are doing
Similar to the user agreement, it would be nice, for tasks that could break things if not done in the right way, to have a step in place that will ask the user to agree that they know what they are doing before allowing them to proceed with links to how they can learn how to do what they are starting the flow for (e.g. making a menu change)

## Points of contact
### Editor end
It would be nice to have somewhere in the KB that we could point editors to for a point of contact for things they can take care of themselves (e.g. contact for VHA Digital Media for PDF rules). It would also be nice if this was in context of the thing they were actively working on (e.g. link in PDF documentation to VHA Digital Media).

### Help desk end
It would be nice to have somewhere in the KB or Confluence space for help desk to reference for a master list of points of contact for who they should contact. It would also be nice if this was in context of the thing they were actively working on (e.g. link in PDF documentation to VHA Digital Media).

## Glossary of terms
It would be nice to have a single location where anyone could find an entire glossary of terms as related to Drupal. Could be like the /wtf bot in slack or just a simple single page with a searchable list of all terms. This could also evolve into something where anytime a term that exists in the Glossary is used in the system, we could have a way for the user to know it’s a term that can be looked up for the definition.

#### Cons:
- Would need to be maintained whenever new terms are added
- Could become lots of dev work down the line if we expand to automate
